File: urbanairship-core/src/test/java/com/urbanairship/job/WorkUtilsTest.java
package com.urbanairship.job;

import com.urbanairship.BaseTestCase;
import com.urbanairship.base.Extender;
import com.urbanairship.json.JsonException;
import com.urbanairship.json.JsonMap;
import com.urbanairship.push.PushManager;

import org.junit.Test;

import java.util.concurrent.TimeUnit;

import androidx.annotation.NonNull;
import androidx.work.Data;

import static org.junit.Assert.assertEquals;

public class WorkUtilsTest extends BaseTestCase {

    @Test
    public void testConvert() throws JsonException {
        JobInfo original = JobInfo.newBuilder()
                                  .setAction("some action")
                                  .setAirshipComponent(PushManager.class)
                                  .setConflictStrategy(JobInfo.APPEND)
                                  .setExtras(JsonMap.newBuilder()
                                                    .put("key", "value")
                                                    .build())
                                  .setMinDelay(10, TimeUnit.MILLISECONDS)
                                  .setNetworkAccessRequired(true)
                                  .setInitialBackOff(10, TimeUnit.SECONDS)
                                  .addRateLimit("foo")
                                  .addRateLimit("bar")
                                  .build();

        Data data = WorkUtils.convertToData(original);
        JobInfo converted = WorkUtils.convertToJobInfo(data);
        assertEquals(original, converted);
    }
}

File: urbanairship-core/src/test/java/com/urbanairship/job/WorkUtilsTest.kt
package com.urbanairship.job

import androidx.test.ext.junit.runners.AndroidJUnit4
import com.urbanairship.job.JobInfo.Companion.newBuilder
import com.urbanairship.job.WorkUtils.convertToData
import com.urbanairship.job.WorkUtils.convertToJobInfo
import com.urbanairship.json.jsonMapOf
import com.urbanairship.push.PushManager
import kotlin.time.Duration.Companion.milliseconds
import kotlin.time.Duration.Companion.seconds
import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
public class WorkUtilsTest {

    @Test
    public fun testConvert() {
        val original = newBuilder()
            .setAction("some action")
            .setScope(PushManager::class.java.name)
            .setConflictStrategy(JobInfo.ConflictStrategy.APPEND)
            .setExtras(jsonMapOf("key" to "value"))
            .setMinDelay(10.milliseconds)
            .setNetworkAccessRequired(true)
            .setInitialBackOff(10.seconds)
            .addRateLimit("foo")
            .addRateLimit("bar")
            .build()

        val data = convertToData(original)
        val converted = convertToJobInfo(data)
        Assert.assertEquals(original, converted)
    }
}
